pyproject.toml

```
@@ -70,6 +71,9 @@ adlfs = { version = ">=2023.1.0,<2024.1.0", optional = true }
 gcsfs = { version = ">=2023.1.0,<2024.1.0", optional = true }
 psycopg2-binary = { version = ">=2.9.6", optional = true }
 sqlalchemy = { version = "^2.0.18", optional = true }
+numpy = { version = "^1.26.3", python = "3.12" }
+greenlet = {version = "^3.0.3", python = "3.12" }
```

Would this be better?
```suggestion
numpy = { version = "^1.26.3", python >= "3.12" }
greenlet = {version = "^3.0.3", python >= "3.12" }
```

---

As we are on the latest version of both the libraries, we should be good with this as upcoming python version support should be there too.